packages/elements/src/components/MosaicTableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,236 @@
+import { Box, Flex, Icon } from '@stoplight/mosaic';
+import * as React from 'react';
+
+import { NODE_META_COLOR, NODE_TYPE_ICON_COLOR, NODE_TYPE_META_ICON, NODE_TYPE_TITLE_ICON } from './constants';
+import {
+  CustomLinkComponent,
+  TableOfContentsDivider,
+  TableOfContentsGroup,
+  TableOfContentsGroupItem,
+  TableOfContentsNode,
+  TableOfContentsNodeGroup,
+  TableOfContentsProps,
+} from './types';
+import {
+  getHtmlIdFromItemId,
+  isDivider,
+  isExternalLink,
+  isGroup,
+  isGroupOpenByDefault,
+  isNode,
+  isNodeGroup,
+} from './utils';
+
+const ActiveIdContext = React.createContext<string | undefined>(undefined);
+const LinkContext = React.createContext<CustomLinkComponent | undefined>(undefined);
+
+export const TableOfContents = React.memo<TableOfContentsProps>(({ tree, activeId, Link }) => {
+  React.useEffect(() => {
+    if (activeId && typeof window !== 'undefined') {
+      const elem = window.document.getElementById(getHtmlIdFromItemId(activeId));
```

Comment:
Perhaps it would be possible to use `ref` or something like that?

Not a must, but would be more `react`y. ;)